### PR TITLE
Wen's weather app

### PR DESCRIPTION
**Pull request by wwenzz**
## Netlify link
https://wen-weather-app.netlify.app/

## Collaborators
[wwenzz]

### CODE REVIEW

**Review by SofieFerrari — COMMENTED** [review]
Wen, as per usual I am super impressed with your coding skills and work structure. I really liked that you fetched both APIs in one function. I also really appreciate reading your code, it is very clear and your comments make it easy to follow.

File: index.html [hunk 1]
```diff
@@ -0,0 +1,77 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <link rel="stylesheet" href="./style.css" />
+    <link rel="preconnect" href="https://fonts.googleapis.com" />
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
+    <link
+      href="https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,100;0,300;0,400;0,500;0,700;0,900;1,100;1,300;1,400;1,500;1,700;1,900&display=swap"
```
Comment (HIPPIEKICK) [inline index.html:10]: Is roboto used?

File: script.js [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    "Expect drizzle today. Bring a light rain jacket or umbrella for added comfort outdoors.",
+  snow: "Snow is expected today. Exercise caution while driving and wear your ice shoes",
+  atmosphere:
+    "Drive safely! Poor atmosphere is reducing visibility on the roads.",
+  rain: "Bring your umbrella today! Stay dry and plan your commute accordingly.",
+  clear:
+    "Enjoy the clear skies today! Get your sunnies on to stay protected from the sunshine.",
+  clouds:
+    "It's a cloudy day. Grab a light jacket and enjoy the cooler temperatures and cozy atmosphere.",
+};
```
Comment (HIPPIEKICK) [inline script.js:22]: Nice!

File: script.js [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+  const minutes = "0" + formattedTime.getMinutes();
+  // get the last two elements in the minutes string, if 08 -> 08 whereas 052 -> 52
+  formattedTime = `${hours}:${minutes.slice(-2)}`;
+  return formattedTime;
+};
+
+// Function that capitalizes the first letter
+const capitalizeFirstLetter = str => {
+  return str.charAt(0).toUpperCase() + str.slice(1);
+};
```
Comment (HIPPIEKICK) [inline script.js:65]: Nice and clear function!

File: script.js [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+
+// Function that returns an array of the weather details for next four days.
+const generateFourDaysWeather = (
+  forecastData,
+  currentDayIndex,
+  currentDate
+) => {
+  let fourDaysWeather = forecastData.list
+    .filter(weather => {
+      return !weather.dt_txt.includes(currentDate);
```
Comment (HIPPIEKICK) [inline script.js:84]: Nice solution!

File: script.js [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+  });
+};
+
+//        Promise-based functions      //
+// Function that fetches geolocation API and returns a promise object
+const getPosition = () => {
+  return new Promise((resolve, reject) =>
+    navigator.geolocation.getCurrentPosition(resolve, reject)
+  );
+};
```
Comment (HIPPIEKICK) [inline script.js:160]: Nice with geolocation 👍

File: script.js [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+  );
+  const fetchForecastAPI = fetch(
+    `${baseURL}${forecastEndpoint}?${queryParam}&units=${queryUnits}&APPID=${apiKey}`
+  );
+  const fetchResponses = Promise.all([fetchWeatherAPI, fetchForecastAPI]);
+  fetchResponses.then(responses => {
+    const weatherRes = responses[0];
+    const forecastRes = responses[1];
+    if (!(weatherRes.ok || forecastRes.ok)) {
+      alert("Location not found, try to search a place accurately.");
```
Comment (HIPPIEKICK) [inline script.js:175]: Nice error handling 🙌

File: script.js [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+  });
+};
+
+////////Event Listener////////
+buttonIcon.addEventListener("click", () => {
+  forecastField.classList.toggle("hide");
+  currentWeatherField.classList.toggle("show");
+  buttonField.classList.toggle("move");
+  weatherAdviceContainer.classList.toggle("show-quote");
+});
```
Comment (HIPPIEKICK) [inline script.js:197]: What a nice addition!

**Review by HIPPIEKICK — APPROVED** [review]
Good job with the weather app! Cute design and a lot of extra features that makes it even nice - like the loader, geolocation and search bar. Keep it up ⭐

File: index.html (hunk 1 above)
Comment (wwenzz) [reply index.html]: Thank you for your positive feedback Matilda! Really appreciate that. I have removed this unused font.

**Pull request closed without merging**